mobile/library/java/io/envoyproxy/envoymobile/engine/HeaderMatchConfig.java

```diff
@@ -0,0 +1,32 @@
+package io.envoyproxy.envoymobile.engine;
+
+import java.util.List;
+
+/* Datatype used by the EnvoyConfiguration to header matches for virtual clusters
+ *
+ * All the fields here map to the fields by the respective names in the HeaderMatcher message
+ * defined here:
+ * https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/route/v3/route_components.proto
+ * */
+public class HeaderMatchConfig {
+  public enum Type {
+    // Indicates the match value is for an exact match.
+    EXACT,
```

Comment:
I would recommend we explicitly specify the int values for each of these cases as otherwise we depend on the order in which these constants are defined in here (changing the order will change underlying integer values which will break the feature).

So let's either add extra documentation that makes it super clear that the order matters in here or - I think it's going to be more future proof - explicitly set the integer value of each of these enum cases so that do not change no matter what (and we can specify arbitrary values as needed).

Reply:
given the way I was converting (based on enum, not value) we don't depend on the values or ordering here.